riscv/triggers.cc

```diff
@@ -461,6 +461,16 @@ bool module_t::tdata3_write(unsigned index, const reg_t val) noexcept
   return true;
 }
 
+bool module_t::icount_enabled() const
```

Comment:
This function is fairly expensive to compute, which slows down the computation of whether the slow path should be taken. Consider precomputing it so that the triggers don’t need to be scanned every time. It’s OK to be conservative and occasionally take the slow path when unnecessary, if that helps.

Reply:
https://github.com/riscv-software-src/riscv-isa-sim/pull/1220/commits/f52de86dc168222264a401e59a7c5e4eef252670 adds a ```check_icount_enabled``` variable to reduce the computation amount of the ```icount_enabled()```. For general cases, the ```check_icount_enabled``` will be false, and the ```icount_enabled()``` will return false directly. The only scenario that requires performing the complete check is when writing tdata1. In this case, the ```check_icount_enabled``` will be set to true, and ```icount_enabled()``` will perform the full check. If the checking is false, we can set the ```check_icount_enabled``` back to false and skip the full check next time.

@aswaterman Is this computation overhead acceptable?

Reply:
Yeah, SGTM.

Reply:
@YenHaoChen there is already a `processor_t::trigger_updated()` method that calculates similar things for fetch/load/store triggers. Why not do the computation and set a variable `check_triggers_icount` (name chosen to be similar to others that already exist) in that method?

Reply:
Good idea, computing `processor_t::trigger_updated()` could compute `processor_t::check_triggers_icount` precisely, superseding the need for `module_t::icount_enabled()`.  That seems a little cleaner.